### PR TITLE
fix(alerts): Unify how alert policy selects an account_id

### LINKED ISSUE #773: account_id not inherited correctly from provider config block
I've been working with Sean Wiley from NewRelic on this issue and he asked me to post it here.

### Terraform Version
Terraform v0.11.11

### Affected Resource(s)
newrelic_alert_policy (data source)

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
terraform {
  required_version = "= 0.12.13"
}

provider "newrelic" {
  version = "2.1.2"
  api_key = "<redacted>
  admin_api_key = "<redacted>"
  account_id = "<redacted>
  region = "US"
}

data "newrelic_alert_policy" "ops_notify_slack" {
  name = "ops-notify-slack"
}
```

### Debug Output

```
$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.newrelic_alert_policy.ops_notify_slack: Refreshing state...

Error: 401 response returned: Bad API Key or no API Key provided, null

  on data.tf line 1, in data "newrelic_alert_policy" "ops_notify_slack":
   1: data "newrelic_alert_policy" "ops_notify_slack" {
```

### Expected Behavior
Provider should authenticate against our account and be able to use API to retrieve data source information.

### Actual Behavior
A 401 error is returned as shown above.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
I realize this is probably not at all reproducible without access to our account and the keys.  Sean Wiley has verified that everything appears to configured correctly per the 2.x migration guide.  We've been using the 1.18.0 provider extensively for some time.

### References
Could not find any

### PR DESCRIPTION
**Pull request by jthurman42**
Uses a consistent function to populate the `account_id` on Alert Policy implementations.

Resolves #773

### CODE REVIEW

File: newrelic/helpers.go [hunk 1]
```diff
@@ -108,3 +94,15 @@ func stringInSlice(slice []string, str string) bool {
 
 	return false
 }
+
+// setIfConfigured will set a resource attribute if it was found to already
+// exist in the configuration for the resource.
+func setIfConfigured(d *schema.ResourceData, attr string, value interface{}) error {
```
Comment (sanderblue) [inline newrelic/helpers.go:100]: Nice addition to the helper functions 😎

File: newrelic/resource_newrelic_alert_policy_test.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// +build integration
```
Comment (ctrombley) [inline newrelic/resource_newrelic_alert_policy_test.go:1]: There are still unit tests in this file marked as `IsUnitTest`.

File: newrelic/structures_newrelic_alert_condition_test.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// +build integration
```
Comment (ctrombley) [inline newrelic/structures_newrelic_alert_condition_test.go:1]: There are still unit tests in this file marked as `IsUnitTest`.

File: newrelic/resource_newrelic_plugins_alert_condition_test.go [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// +build integration
```
Comment (ctrombley) [inline newrelic/resource_newrelic_plugins_alert_condition_test.go:1]: There are still unit tests in this file marked as `IsUnitTest`.

File: website/docs/r/alert_policy.html.markdown [hunk 5]
```diff
@@ -27,7 +27,7 @@ The following arguments are supported:
   * `name` - (Required) The name of the policy.
   * `incident_preference` - (Optional) The rollup strategy for the policy.  Options include: `PER_POLICY`, `PER_CONDITION`, or `PER_CONDITION_AND_TARGET`.  The default is `PER_POLICY`.
   * `channel_ids` - (Optional) An array of channel IDs (integers) to assign to the policy. Adding or removing channel IDs from this array will result in a new alert policy resource being created and the old one being destroyed. Also note that channel IDs _cannot_ be imported via `terraform import` (see [Import](#import) for info).
-  * `account_id` - (Optional) The New Relic account ID to operate on.  This allows the user to override the `account_id` attribute set on the provider.
+  * `account_id` - (Optional) The New Relic account ID to operate on.  This allows the user to override the `account_id` attribute set on the provider. Defaults to the environment variable `NEW_RELIC_ACCOUNT_ID`.
```
[the remark this answers is not in the document]
Comment (sanderblue) [reply website/docs/r/alert_policy.html.markdown:30]: Yeah, maybe that's better wording.

**Pull request merged**